lib/puppet/resource_api/transport_schema_def.rb

```diff
@@ -0,0 +1,8 @@
+require 'puppet/resource_api/base_type_definition'
+# Provides accessor methods for the type being provided
+class Puppet::ResourceApi::TransportSchemaDef < Puppet::ResourceApi::BaseTypeDefinition
+  def initialize(definition)
+    raise Puppet::DevError, 'TransportSchemaDef requires definition to be a Hash' unless definition.is_a?(Hash)
```

Comment:
duplicate check - if you're doing this because of the class name in the error message, use `self.class.name` in the base class.